SM-35 product card

n0macx opened the pull request:
`ProductPage with ProductCards. Mock data and fetch and modal`

![image](https://user-images.githubusercontent.com/66994199/167680347-d6749ac6-ee61-49e4-ab7b-384045216163.png)
![image](https://user-images.githubusercontent.com/66994199/167680815-5cf538cc-ed34-4824-9a1f-7dce6d1021b3.png)
![image](https://user-images.githubusercontent.com/66994199/167680897-9a238e45-4891-4609-8c45-f4f8e6626b1e.png)
![image](https://user-images.githubusercontent.com/66994199/167680983-2e8a1395-5bd8-4425-b3e0-08bb3438fe8e.png)

n0macx committed:
SM-35 Added: Eslint rule and prettier minor fix

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
SM-35 Refresh

n0macx committed:
SM-35 Tailwind trials

n0macx committed:
SM-35 Added Tailwind prettier plugin

n0macx committed:
SM-35 Added: Mock img and container styles

n0macx committed:
SM-35 Added: 	ProductCard component:
		- Styled and ready for logic implementation
		index.html, index.css:
		- Inter google font added

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
SM-35 App after merge

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
SM-33 Work in progress

n0macx committed:
SM-35 Layout fixed, ProductCard component styled responisively

n0macx committed:
SM-35 Products Page with mock data and fetch
	Added:	- ProductsPage with mapped product cards and mock fetch (TODO: gathering data from redux store?)
		- mockProductData separate file

n0macx committed:
SM-35 cleaned

n0macx committed:
SM-33 Added: series description and title

n0macx committed:
SM-35 cleaned

n0macx committed:
SM-35 Added: logo carpet

n0macx committed:
SM-35 Added Modal Mock

n0macx committed:
SM-35 Language translations added

n0macx committed:
SM-35 Added: alt text from mock object

MonikaKrella commented on src/assets/mockData/mockPoductData.tsx:
```diff
@@ -0,0 +1,42 @@
+    price: 75,
+    productImage: `${mockImg}`,
+    altText: 'single mallet',
+    seriesId: 'Beethoven Series',
```
seriesId will be Id to series table, but it is easy to join here full series object or jus property seriesName

n0macx replied:
Resolved

MonikaKrella commented on src/assets/mockData/mockPoductData.tsx:
```diff
@@ -0,0 +1,42 @@
+    headDiameter: 43,
+    stickLength: 37.5,
+    weight: 34,
+    price: 75,
```
will be headDiameter, weight or pirce as floats? or they will be intigers?

n0macx replied:
Floats mostly.

RobertS-ki commented on src/features/products/ProductCard.tsx:
```diff
@@ -0,0 +1,84 @@
+            </div>
+            <p className="text-lg sm:pr-20 sm:text-xl">
+              {prop.productObject.productDescription}
+            </p>
```
No translation for product description?

n0macx replied:
Added. Thanks !

RobertS-ki replied:
I ment opacity.

RobertS-ki commented on src/features/products/ProductCard.tsx:
```diff
@@ -0,0 +1,84 @@
+              <button
+                className="mt-10 h-12 w-56 rounded-md bg-[#232323] px-6 font-bold uppercase text-white shadow-none transition-shadow duration-300 
+          ease-in-out hover:shadow-xl sm:mt-0"
+              >
```
I barely see the shadow under the button.

n0macx replied:
Shadow is big enough. Maybe later we decide to leave it or not.

RobertS-ki replied:
I ment opacity.

RobertS-ki commented on src/features/products/ProductCard.tsx:
```diff
@@ -0,0 +1,84 @@
+          </h1>
+          <Backdrop handleClose={handleDetailsModal} />
+        </>
+      )}
```
It always displays in the middle of the page - mock details should display on top of the selected product (or be easily related). TODO?

n0macx replied:
Good Point! Its just a mock though for easier implementation.

n0macx committed:
SM-35 Major style changes

n0macx committed:
SM-35 Refactored Mock data in case of language changes

n0macx committed:
SM-35 Refactored mocks to adapt for proper database data
	Added: 	- productTypes - exported
		- fetch od products Page takes full series object
		- destructure porps in productCard

vieraboschkova commented on src/App.tsx:
```diff
@@ -30,6 +31,7 @@ function App() {
               <Route path="/about" element={<AboutMe />} />
               <Route path="/contact" element={<ContactPage />} />
               <Route path="/product-series" element={<ProductSeriesPage />} />
+              <Route path="/products-page" element={<ProductsPage />} />
```
Check links in navbar and footer if they guide to correct links

n0macx replied:
Refactored. Thanks

vieraboschkova replied:
I think there is still a bug somewhere, link from footer point to product-series (not products). 
Also, i think you might need to add a param to url -> products/beethoven, products/bartok, and then if you have a specific type, products/bartok/bm1
(my suggestion for refactoring url: series/bartok/bm1)

n0macx replied:
Think we will do it when making series page

vieraboschkova commented on src/features/products/ProductsPage.tsx:
```diff
@@ -0,0 +1,79 @@
+];
+
+const ProductsPage = () => {
+  const { i18n } = useTranslation('navAndFooter');
```
You may want to create a different file for translations for this page

vieraboschkova commented on src/features/products/ProductsPage.tsx:
```diff
@@ -0,0 +1,79 @@
+
+  return (
+    <>
+      {pending && <h1 className="text-center">Loading...</h1>}
```
It's jumping in between the mocked "loading" but I guess it will be something to fix once the fetch from db will be performed

n0macx replied:
Yes, its prepared for implementation of Loader and then we will fix it.

vieraboschkova commented on src/features/products/ProductsPage.tsx:
```diff
@@ -0,0 +1,79 @@
+      <div className="py-4 px-12">
+        <h1 className="py-2 text-2xl font-bold">{seriesMock.seriesName}</h1>
+        <h2 className="max-w-3xl py-2 text-xl">
+          {i18n.language === 'en' ? seriesLang.en : seriesLang.pl}
```
seriesLang[i18n.language] should do the trick also

n0macx replied:
In this case i would prefer this solution.

KatarzynaChmielecka replied:
> seriesLang[i18n.language] should do the trick also

for now it's not working :(

vieraboschkova commented on src/features/layout/footer/Footer.tsx:
```diff
@@ -7,7 +7,7 @@ const Footer = () => {
   return (
     <>
-      <footer className=" h-fit w-full bg-[#232323] text-white">
+      <footer className="w-full bg-[#232323] text-white">
```
when I click on one of the footers links, the scroll seems to stay on the bottom. Check and see if maybe a (custom) useScrollToTop hook could be useful

n0macx replied:
Good point. I think we will make a scroll button globally, but i will try to make a hook also.

vieraboschkova replied:
Yeah, scroll to top is a nice feature, but when you enter new page you expect to be on the top of it usually (unless explicitly not :) )

n0macx replied:
Done.

n0macx committed:
SM-35 Fixed links and added mock link

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
SM-35 layout fixed

n0macx committed:
SM-35 layout fix

n0macx committed:
SM-35 Scroll To Top added after changing pathname

n0macx committed:
SM-35 Added Tools folder

n0macx committed:
SM-35 ScrollTOTOp component moved

n0macx committed:
SM-35 type o fix

n0macx committed:
Merge branch 'development' into SM-35-product-card

n0macx committed:
SM-35 Fixed useTranslation parameters

n0macx committed:
SM-35 Loader added

n0macx committed:
SM-35 final amendments

n0macx committed:
SM-35 Last microsection

KatarzynaChmielecka reviewed (approved):
Well done! :)
